### PR TITLE
Initial Android Support

### PR DESCRIPTION
**Pull request by donnfelker**
Adding Android support
    
New path configuration file for Android.
SVG Files since Android doesnt ship with any.
Enable authentication on destroy action of auths controller.
Removed iOS native registration in path configuration in as we do not have a user controller in the API. RE: #15 

Still to do (in another PR: Notifications)

Supports #13

### CODE REVIEW

File: app/controllers/turbo/android/path_configurations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+              }
+            },
+            {
+              patterns: ["/users/sign_in"],
+              properties: {
+                uri: "turbo://fragment/users/sign_in",
+                context: "modal"
+              }
+            },
+            # Needs a /api/v1/users_controller.rb for auth if you want native sign up
```
Comment (donnfelker) [inline app/controllers/turbo/android/path_configurations_controller.rb:53]: Native registration is disabled until (if/when) we get a /api/v1/users_controller

File: app/controllers/turbo/android/path_configurations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            tabs: [
+              {
+                title: "Home",
+                path: root_path,
+                image_url: svg_icon_path("home")
+              },
+              {
+                title: "Posts",
+                path: posts_path,
+                image_url: svg_icon_path("compose")
```
[the remark this answers is not in the document]
Comment (joemasilotti) [reply app/controllers/turbo/android/path_configurations_controller.rb:17]: I worry that adding a route that will 404 might confuse folks. I vote we either:

1. Generate a scaffold for a Posts route so this works
2. Remove this tab and only use Home

If we go with 1. then we should also add the route to the iOS so the tabs match.

File: app/controllers/turbo/android/path_configurations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+              }
+            },
+            # TODO: Needs a /api/v1/users_controller.rb for registration to work
+            # {
+            #   patterns: ["/users/sign_up"],
+            #   properties: {
+            #     uri: "turbo://fragment/users/sign_up",
+            #     context: "modal"
+            #   }
+            # }
```
Comment (joemasilotti) [inline app/controllers/turbo/android/path_configurations_controller.rb:60]: I vote we remove these comments and add a new issue to show that it's on the roadmap.

Comment (joemasilotti) [reply app/controllers/turbo/android/path_configurations_controller.rb:60]: FWIW, I can add the registration flow later next week.

File: app/controllers/turbo/android/path_configurations_controller.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (donnfelker) [reply app/controllers/turbo/android/path_configurations_controller.rb:17]: Why would this 404? This example has this route. 

If there is only one tab, the tab bar is not shown. Hence the need for two tabs.

File: app/controllers/turbo/android/path_configurations_controller.rb (hunk 3 above)
Comment (donnfelker) [reply app/controllers/turbo/android/path_configurations_controller.rb:60]: Sounds good. I'll add an issue and remove the commented out code.

Comment (donnfelker) [reply app/controllers/turbo/android/path_configurations_controller.rb]: #15 created, and commented out code has been removed.

Comment (joemasilotti) [reply app/controllers/turbo/android/path_configurations_controller.rb]: Awesome, thanks!

File: app/controllers/turbo/android/path_configurations_controller.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (joemasilotti) [reply app/controllers/turbo/android/path_configurations_controller.rb:17]: Ah, my mistake! You're absolutely right, `/posts` exists. I'll update the iOS Path Configuration to match.

**Pull request merged**